chainer/links/caffe/caffe_function.py

```diff
@@ -435,3 +484,135 @@ def __init__(self, caffe_func, name):
 
     def __call__(self, *xs):
         return self.caffe_func[self.name](*xs)
+
+
+class _EltwiseFunction(object):
+    def __init__(self, operation, coeffs=None):
+        if coeffs is not None:
+            assert len(coeffs) > 0
+        self.operation = operation
+        self.coeffs = coeffs
+
+    def __call__(self, *xs):
+        operation = self.operation
+
+        if operation == 0:      # PROD
+            return reduce(lambda x, y: x * y, xs),
+
+        elif operation == 1:    # SUM
+            coeffs = self.coeffs
+            if coeffs is not None:
+                assert len(xs) == len(coeffs)
+                xs2 = [x * coeff for x, coeff in zip(xs, coeffs)]
+            else:
+                xs2 = xs
+            return reduce(lambda x, y: x + y, xs2),
+
+        elif operation == 2:    # MAX
+            return reduce(lambda x, y: functions.maximum(x, y), xs),
+
+        else:
+            raise RuntimeError('Invalid EltwiseParameter.EltwiseOp value.')
+
+
+class _ScaleFunction(object):
+    def __init__(self, axis):
+        self.axis = axis
```

Comment:
Scale Layer in Caffe supports negative `axis`, and normalize it by [Blob::CanonicalAxisIndex](https://github.com/BVLC/caffe/blob/be163be0ea5befada208dbf0db29e6fa5811dc86/include/caffe/blob.hpp#L118), so it is preferable to implement it or add comments that we do not support negative axes.